static/js/rich_search/app.tsx

```diff
@@ -33,35 +33,69 @@ import {
   setTokensToUrl,
 } from "../tools/timeline/util";
 import { getPlaceNames } from "../utils/place_utils";
-import { MemoStatVars, StatVarsPropType } from "./stat_vars";
+import { StatVars, StatVarsPropType } from "./stat_vars";
 
 interface AppPropType {
   query: string;
-  setQuery: (string) => void;
   places: string[];
   addPlace: (string) => void;
   removePlace: (string) => void;
   loading: boolean;
-  onClickSearch: () => void;
-  inputInvalid: boolean;
+  onSearch: (string) => void;
   chartsData: StatVarsPropType;
 }
 
+interface TextSearchBarPropType {
```

Comment:
Please add each new component into a separate file for easier management, and some documentation.